src/hooks/useCredentialApi.ts

```diff
@@ -1,44 +1,170 @@
-  };
+type ApiDeleteCredentialErrorType = {
+  detail?: string;
+  message: string;
+};
+
+const useDeleteCredentialApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (ids: CredentialType['id'][]): Promise<AxiosResponse<ApiDeleteCredentialSuccessType>> =>
+      axios.post(`${process.env.REACT_APP_CREDENTIALS_SERVICE_BULK_DELETE}`, { ids }),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (
+      response: AxiosResponse<ApiDeleteCredentialSuccessType>,
+      updatedCredentials: CredentialType[]
+    ) => {
+      const { data } = response;
+
+      const deletedNames = data?.deleted
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const skippedNames = data?.skipped
+        ?.map(({ credential }) => updatedCredentials.find(cred => credential === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingNames = data?.missing
+        ?.map(id => updatedCredentials.find(cred => id === cred.id)?.name)
+        .filter(Boolean)
+        .join(', ');
+
+      const missingCredsMsg = 'The following credentials could not be found:' + missingNames;
+
+      // All credentials were deleted successfully
+      if (data?.deleted?.length === updatedCredentials.length) {
+        const credentialCount = updatedCredentials.length;
+        const context = credentialCount === 1 ? 'deleted-credential' : 'deleted-credentials';
+        const successMessage = t('toast-notifications.description', {
+          context: context,
+          name: updatedCredentials.map(({ name }) => name).join(', ')
+        });
+        onAddAlert({
+          title: successMessage,
+          variant: 'success',
+          id: helpers.generateId()
+        });
+        return;
+      }
+
+      // No credentials deleted, either all skipped or all missing
+      if (
+        data?.skipped?.length === updatedCredentials.length ||
+        data?.missing?.length === updatedCredentials.length
+      ) {
+        if (data?.skipped?.length === updatedCredentials.length) {
+          const countValues = data?.skipped?.length === 1 ? 'Credential' : 'Credentials';
+          const errorMessage = t('toast-notifications.description', {
+            context: 'skipped-credentials',
+            name: skippedNames,
+            credential_count: countValues
+          });
+          onAddAlert({
+            title: errorMessage,
+            variant: 'danger',
+            id: helpers.generateId()
+          });
+          return;
+        }
+        {
+          console.log(missingCredsMsg);
+          return;
+        }
+      }
+
+      // Some credentials deleted, some skipped or missing
+      if (data?.deleted?.length && data?.skipped?.length) {
+        const skippedCountValues = data?.skipped?.length === 1 ? 'Credential' : 'Credentials';
```

Comment:
required: move to use `count` in i18next. you can display `{{count}}` in your strings too!